dracut/modules.d/90kiwi-dump/kiwi-ramdisk-deployment-generator.sh

```diff
@@ -11,6 +11,8 @@ root_uuid=$(
     grep root= | cut -f2- -d=
 )
 
+[ -z "${root_uuid}" ] && exit 1
+
```

Comment:
> under which circumstances can it happen that we don't have a root variable in config.bootoptions ? or is this sanity check just added for convenience ?

There shouldn't be any circumstance were we see this situation. However the insane boot I was debugging that triggered all this code was getting to this situation because of the missing crypt dependency in `kiwi-lib` (we already merged a fix for that in #1174) was causing an unmanaged script failure before even validating the existence of `config.bootoptions`. Dracut was never complaining of a missing config.bootoptions and the generator produced a wrong unit-file. This wrong unit-file causes systemd to fail on switch-root complaining of a worng unit file format.

I don't see any way now to get to a missing `config.bootoptions`. And my tests playing the stupidity role always ended up with a proper dialog with a `config.bootoptions` file missing message and a reboot button.

I opted to have this sanity check since the generator is overwriting the unit file generated by systemd, thus in the worst case that would be used. In bsc#1142899 they attempted a similar patch that _solved_ (log were still showing many failures of kiwi dracut modules but system was booting) the issue.

Reply:
Ah ok, thanks that makes sense :+1: